cardano-testnet/test/cardano-testnet-test/cardano-testnet-test.hs

```diff
@@ -5,94 +5,92 @@ module Main
   ) where
 
 import qualified Cardano.Crypto.Init as Crypto
-import qualified Cardano.Testnet.Test.Cli.Babbage.LeadershipSchedule
-import qualified Cardano.Testnet.Test.Cli.Babbage.StakeSnapshot
-import qualified Cardano.Testnet.Test.Cli.Babbage.Transaction
-import qualified Cardano.Testnet.Test.Cli.Conway.DRepRetirement as DRepRetirement
-import qualified Cardano.Testnet.Test.Cli.Conway.Plutus
-import qualified Cardano.Testnet.Test.Cli.KesPeriodInfo
-import qualified Cardano.Testnet.Test.Cli.Queries
-import qualified Cardano.Testnet.Test.Cli.QuerySlotNumber
-import qualified Cardano.Testnet.Test.FoldBlocks
-import qualified Cardano.Testnet.Test.LedgerEvents.Gov.ProposeNewConstitution
-import qualified Cardano.Testnet.Test.LedgerEvents.Gov.ProposeNewConstitutionSPO as LedgerEvents
-import qualified Cardano.Testnet.Test.LedgerEvents.SanityCheck as LedgerEvents
-import qualified Cardano.Testnet.Test.LedgerEvents.TreasuryGrowth as LedgerEvents
-import qualified Cardano.Testnet.Test.Node.Shutdown
 import qualified Cardano.Testnet.Test.SubmitApi.Babbage.Transaction
 
 import           Prelude
 
+import qualified Control.Concurrent.QSem as IO
 import qualified System.Environment as E
+import qualified System.IO as IO
 import           System.IO (BufferMode (LineBuffering), hSetBuffering, hSetEncoding, stdout, utf8)
+import qualified System.IO.Unsafe as IO
 
 import qualified Testnet.Property.Run as H
 
 import qualified Test.Tasty as T
 import           Test.Tasty (TestTree)
 import qualified Test.Tasty.Ingredients as T
 
+sem :: IO.QSem
+sem = IO.unsafePerformIO $ IO.newQSem 1
+{-# NOINLINE sem #-}
```

Comment:
It doesn't have to be in `unsafePerformIO`. You have IO access in the usage site below.